apps/bfd-model/bfd-model-rif-samples/dev/synthea-test-plan.md

```diff
@@ -0,0 +1,235 @@
+# Synthea Test Plan
+
+This document details various tests/procedures to ensure that each batch of synthetic data generated will not collide with previously generated data.
+
+## Before Generating Data
+
+1. Check the output of the end-state properties from the latest synthea release i.e. `apps/bfd-model/bfd-model-rif-samples/dev/synthea_releases/XX-XX-20XX/end_state.properties`. 
+2. Query ranges of the end-state properties in PROD SBX to ensure there will not be any overlaps based on the synthetic data set size. e.g.: `psql -h {prod sbx db url} -d fhirdb -U bfduser -f apps/bfd-model/bfd-model-rif/src/main/resources/db/scripts/Query_synthetic_id_ranges.sql | tee Query_synthetic_id_rangesTest.out`.
+3. Increment the end-state properties values by 1 or more based on the query output in step 2 and set the values in the synthea.properties file, and generate new data.
+
+## Loading Data
+
+1. Test CCWRIFLoader functionality locally (i.e. verify BFD Pipeline can ingest records). Copy the files that were generated from the steps above and place them in the rif-synthea folder located at: `apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-synthea`
+2. Go to `/apps/bfd-model/bfd-model-rif-samples/src/main/java/gov/cms/bfd/model/rif/samples/StaticRifResource.java`. Go to and navigate to the SAMPLE_SYNTHEA entries and add or delete the entries that are currently there to correspond to the files you have copied from step 1.  You will also need to know how many records are in each file and place that in the entry as well. Sample Below: 
```

Comment:
Might want to give an example (i.e., hot-to) on getting # of rcds in a given RIF file; somethlng like:

`tail ~/dev/synthetic_data/synthetic-beneficiary-2014.rif | wc -l`

Obviously change the directory and RIF filename as needed; also, might want to mention that the above count will yield one more than the actual rcd count since the "wc -l" will include the header line.